pkg/publicapi/apimodels/agent.go

```diff
@@ -0,0 +1,32 @@
+package apimodels
+
+import "github.com/bacalhau-project/bacalhau/pkg/models"
+
+// IsAliveResponse is the response to the IsAlive request.
+type IsAliveResponse struct {
+	BaseGetResponse
+	Status string
+}
+
+func (r *IsAliveResponse) IsReady() bool {
+	if r != nil && r.Status == "OK" {
```

Comment:
if an error is returned when calling `client.Agent().Alive()`, then the returned response will be `nil`. This check is just safer if the user calls `res.IsReady()` without checking the error.

```
isAliveResponse, err := client.Agent().Alive()

// this will return false instead of nil pointer exception if an error was returned
if isAliveResponse.IsReady() {
  ....
} 
```